src/store/Reducer.ts

```diff
@@ -0,0 +1,61 @@
+import { Reducer } from "react";
+import { AsyncActionHandlers } from "use-reducer-async";
+
+import { State } from "./State";
+import * as Actions from "./Actions";
+import { admin } from "./Textile";
+
+export const reducer: Reducer<State, Actions.Action> = (
+  state,
+  action
+): State => {
+  switch (action.type) {
+    case Actions.InnerType.ErrorFetch:
+      return {
+        ...state,
+        loading: false,
+        hub: { count: 0 }, // TODO: Remove these fake values
+      };
+    case Actions.OuterType.Increment:
+      return {
+        ...state,
+        hub: { count: (state.hub?.count ?? 0) + 1 },
+      };
+
+    // Real Textile APIs
+    case Actions.InnerType.StartSignUp:
+      return {
+        ...state,
+        loading: true,
+      };
+    case Actions.InnerType.FinishSignUp:
+      const { sessionInfo } = action;
+      return {
+        ...state,
+        loading: false,
+        user: { ...state.user, sessionInfo },
+      };
+    default:
+      throw new Error("unknown action type");
+  }
+};
+
+export const asyncActionHandlers: AsyncActionHandlers<
```

Comment:
Here's our async reducer handler. This is just an object with async handlers for our various async action types. We can easily add more actions by adding more keys based on action types. To make things behave nicely, its best to use the promise pattern here (rather than try/cat and async/await). But technically, async/await patterns would also work here. I think things get slightly more complicated when dealing with routing etc.